### ISSUE TITLE
Faster server initialization

### ISSUE THREAD

**Issue by castwide** [castwide/solargraph#415]
The `fast-initialize` branch aims to improve startup speed by allowing the `initialize` message to complete without mapping the entire workspace. The mapping will occur in the background instead.

To do:
- [x] Use LSP progress reporting to track the status of the mapping process
- [x] Avoid raising `FileNotFoundError` when trying to access files that have not been mapped yet

**Comment by castwide** [comment]
Two updates to the language server have been pushed to the master branch:

* The server responds to the `initialize` method without needing to map the workspace first. This keeps clients from assuming the server timed out when mapping is a long-running process. After initialization, the workspace gets mapped in the background
* Clients that support `window/workDoneProgress` will receive progress reports on the mapping process. Screenshot from VS Code:
![mapping-workspace-demo](https://user-images.githubusercontent.com/3704922/120074686-fdfa3900-c06b-11eb-8b22-5b1ed72915f3.png)

These changes will be included in the next versions of the gem and the VS Code extension.

**Comment by castwide** [comment]
Updates published in v0.41.0.

**Comment by gencer** [comment]
I'm not sure If my issue is due to this change or something else but I lost almost entire bundled gem autocompletion.

For example, Let say I have a Rails app that contains many gems. Name it. Like; `octokit`,  `sanitize`, `semantic_logger` or `puma` and `unicorn`...

None of them showes up on autocomplete list:

```ruby
Octokit:: # <- Not even Octokit found
Sani # <- should see Sanitizer in list.
# or
Pum # <- should also see Puma and its members.
```

Except few `Rails` autocompletion, I almost lost every single gem from my workspace when they are installed.

Tried `solargraph clear` and also `bundle` but still no autocomplete for any of my gems. I also have this `.solargraph.yml` in my root:


```yaml
include:
  - '**/*.rb'
exclude:
  - spec/**/*
  - test/**/*
  - vendor/**/*
  - '.bundle/**/*'
require:
  - rails/all
  - actioncable
  - actionmailer
  - actionpack
  - actionview
  - activejob
  - activemodel
  - activerecord
  - activestorage
  - activesupport
  - grape
  - grape-entity
  - sentry
  - sequel
  - countries
  - doorkeeper
  - devise
  - bunny
  - stronger_parameters
  - secure_random
  - faker
  - twilio
  - octokit
  - capistrano
  - pathname
  - uuid4
  - iyzipay
  - zammad_api
  - ruby_identicon
  - dnsruby
  - rugged
  - jwt
  - omniauth
  - redis
  - rack-attack
  - sequel-devise
  - sidekiq
  - countries
  - nexmo
  - prawn
  - letter_avatar
  - uri
  - rethinkdb
  - rails_semantic_logger
  - semantic_logger
domains: []
reporters:
  - rubocop
  - require_not_found
plugins:
  # - solargraph-rails
  - runtime
max_files: 0
```

**Comment by castwide** [comment]
@gencer Confirmed. The maps aren't picking up your .solargraph.yml configuration. If you add the line `require 'octokit'` to one of your Ruby files, it should start working.

I'll fix the problem in the next release.

**Comment by gencer** [comment]
@castwide Thank you for your fast reply. Yes, requiring any gem in `rb` file directly (like you said) works on my end too.

Awaiting for the next release. ❤

**Comment by castwide** [comment]
Fixed in 0.42.0.
